Configurator

braunms opened the pull request:
Added some more features and fixed bugs. See #539
Improved usability.

braunms committed:
added interface slots for groups

braunms committed:
add interface slot fix

braunms committed:
minot changes

braunms committed:
interface slots - wip

braunms committed:
Merge branch 'configurator' of https://github.com/braunms/megamol into configurator

braunms committed:
wip: interface slots

braunms committed:
wip: interface slots ...

braunms committed:
some work on selection ...

braunms committed:
...

braunms committed:
...

braunms committed:
formatting

braunms committed:
minor stuff fixed ..

braunms committed:
minor stuff fixed ..

braunms committed:
wip and minor stuff fixed ..

braunms committed:
minor stuff and wip of interface slots

braunms committed:
Merge branch 'configurator' of https://github.com/braunms/megamol into configurator

braunms committed:
clang formatting and windows compile check + compile fix

braunms committed:
interface ptr fix

braunms committed:
merge

braunms committed:
wip

braunms committed:
some cleanup of interface slots stuff ...

braunms committed:
added parameter presentation mode

braunms committed:
windows compile check and clang formatting

braunms committed:
fixed imgui push pop bug, finished parameter presentation for pinning value to mouse pos

braunms committed:
re-added module filtering

braunms committed:
minor stuf ...

braunms committed:
readme

braunms committed:
readme, header shuffle, saving gui state of parameters to gui state parameter

braunms committed:
formatting

braunms committed:
working on configurator state

braunms committed:
fixed interface slot bug

braunms committed:
graph button stuff: started fixing ...

braunms committed:
graph button stuff: started fixing ...

braunms committed:
next try

braunms committed:
...

braunms committed:
...

braunms committed:
finally fixed button overlay interaction

braunms committed:
wip interface slots

braunms committed:
wip interface slots

braunms committed:
clang foramtting and win compile check

braunms committed:
wip interface slot ... only drop action missing

braunms committed:
imsc

braunms committed:
imsc

braunms committed:
interface slot for connection within group works

braunms committed:
fix

braunms committed:
fix ... it seems to work

braunms committed:
added other dnd cases for interface slots ... and fixes ofcourse

braunms committed:
improved drawing order of graph elements, fixed group selection, readne, further work on save/load of graph and configurator state

braunms committed:
interface restore fix

braunms committed:
fixes and work on save/load state ...

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
save/load state is working

braunms committed:
interface slot loading

braunms committed:
windows compile test and clang formatting

braunms committed:
fixes ...

braunms committed:
windows compile test and clang formatting

braunms committed:
demo pic

braunms committed:
added interface label, new module group membership, group view toggle on header double-click

braunms committed:
fix broken module deletion

braunms committed:
allow deletion of 'connceted' interface slots

braunms committed:
wip: layouting considering groups, minor changes ...

braunms committed:
improved layouting

braunms committed:
minor code restructuring

braunms committed:
minor fix

braunms committed:
fixed inconsistent interface slot removal/addition

braunms committed:
fixes ...

braunms committed:
ndemodule placement, layouting wip

braunms committed:
hotkey scoping

braunms committed:
wip layouting

braunms committed:
wip layouting

braunms committed:
ignore cycles in layout

braunms committed:
...

braunms committed:
added layouting for selected modules and module withi groups, fixed dnd, compatible highlighting, deletion via context menu

braunms committed:
minor rework of slot compatibility checks

braunms committed:
minor rework of slot compatibility checks

braunms committed:
continued rework of slot compatibility checks ...

braunms committed:
fixed segfault ...

braunms committed:
interface deletion triggers selective call deletion

braunms committed:
misc

braunms committed:
misc

braunms committed:
misc

braunms committed:
readme

braunms committed:
better parameter indentation

braunms committed:
...

braunms committed:
clang and windows compile check

braunms committed:
disabled verbose output

braunms committed:
...

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
using imgui::caltextsize instead own impl

braunms committed:
added edit files for flexenum params

braunms committed:
minor change

braunms committed:
added keyboard focus to text input for new flex enum value

braunms committed:
merge

braunms committed:
clang formatting and windows compile test

braunms committed:
some minor gui fixes

braunms committed:
saving additional tf editor options in state

braunms committed:
moved pop-ups to separate function

braunms committed:
separated and unified param gui presentation definition for gui and configurator

braunms committed:
core api fix

braunms committed:
core api fix, cont.

braunms committed:
disabled vertcal spacing between modules

braunms committed:
separated parameter presentations

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
dragged presentation class out of graph classes, parameter presentation

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
WIP: merge duplicate parameter widget code

braunms committed:
WIP ...

braunms committed:
added permanent menu bar, moved saving configurator project to graph tab, added saving of additional gui settings, minor refactoring of window manager

braunms committed:
tf editor uses gui interanl parameters now

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
clang formatting and windows compile check

braunms committed:
minor fix

braunms committed:
minor fix

braunms committed:
again minor appveyor fix

braunms committed:
fixed parameter synchronization

braunms committed:
Merge branch 'configurator' of https://github.com/braunms/megamol into configurator

braunms committed:
param stuff

braunms committed:
WIP: replacing core instance dependency ...

braunms committed:
minor stuff

braunms committed:
merge

braunms committed:
WIP: Loading running graph for gui

braunms committed:
merge

braunms committed:
WIP: Loading running graph for gui ...

braunms committed:
reducedchanged interface to presentation classes

braunms committed:
new graph classes interface

braunms committed:
completed graph loading for gui

braunms committed:
wip save load mess ...

braunms committed:
reverted changes of project saving

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
fixed transfer fucntion texture preview

braunms committed:
finalized core graph replacement in gui

braunms committed:
minor stuff

braunms committed:
param gui option fix

braunms committed:
added empty param widget group class

braunms committed:
added empty param widget group class ...

braunms committed:
toggle local and global tf editor

braunms committed:
revert

braunms committed:
some UI fixes, wip: parameter group ...

braunms committed:
revert ...

braunms committed:
...

braunms committed:
minor ui fixes, added first version of animation group widget

braunms committed:
added simple grouped widget for animation

braunms committed:
formatting and include fix

braunms committed:
some cleanup, added gui state options to param group

braunms committed:
removed namespace configurator::

braunms committed:
moved files and renamed folder

braunms committed:
further restructuring, removed saving using core instance

braunms committed:
fixed state saving

braunms committed:
fixed state loading

braunms committed:
comment

braunms committed:
added gui state menu to parameter group

braunms committed:
added knob widget for float and int (wip)

braunms committed:
knob widget ...

braunms committed:
removed int knob

braunms committed:
moved docu to readme, minor widget changes

braunms committed:
knob widget and readme

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
typo

braunms committed:
added additonal image and fixed breaking windows loop

braunms committed:
gui image

invor commented:
Played around with current state (tested in CompositingExample.lua). Tow minor problems:
- File dialog of GlTFFileLoader will not go outside MegaMol root directory with folder up command
- deleting a module inside a group will create a new interface slot for that call. Is that on purpose?

braunms committed:
changed default presentation of animation group

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
fixed relative path stalling, removed needless additional interface slots after module deletion

braunms commented:
Fixed minor problems from above in d2532de:
- Relative path ".." blocks going to parent directory. Only ".." is now converted to absolute path, keeping possibility of using relative paths in file path parameters.
- Adding new interface slots on module deletion was not on purpose :-P

braunms committed:
formatting, minor changes

braunms committed:
moved widgets to separate files, added image widget using texture - wip

braunms committed:
wip

braunms committed:
wip image widget

braunms committed:
added empty picking file, still some minor refactoring

braunms committed:
tf tex uv fix

braunms committed:
pasted commented code for picking

braunms committed:
reference cleanup fix and hotkey check fix

braunms committed:
wip picking (disabled for merge to master)

braunms committed:
Merge branch 'master' of https://github.com/UniStuttgart-VISUS/megamol into configurator

braunms committed:
adapted interface slot visibility

braunms committed:
fixed interface slot label width, fixed interface slot position for d&d, fixed inconsistent interface slot restoration

invor reviewed (approved):
Tested loading and editing different projects. Everything seems fine so far.
